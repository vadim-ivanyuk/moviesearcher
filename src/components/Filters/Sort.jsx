import React from "react";
import { useDispatch } from "react-redux";
import { onChangeFilters } from "../../redux/filters/filters.actions";

const sortByOptions = [
  {
    label: "Популярные (убывание)",
    value: "popularity.desc",
  },
  {
    label: "Популярные (возростание)",
    value: "popularity.asc",
  },
  {
    label: "Рейтинг (убывание)",
    value: "vote_average.desc",
  },
  {
    label: "Рейтинг (возростание)",
    value: "vote_average.asc",
  },
];

export const Sort = () => {
  const dispatch = useDispatch();

  const handleChange = (e) => {
    const { name, value } = e.target;

    dispatch(onChangeFilters(name, value));
  };

  return (
    <div className="sort form-group">
      <label className="filters-label" htmlFor="sort_by">
        Сортировать
      </label>
      <select
        className="form-control cursor-pointer"
        name="sort_by"
        id="sort_by"
        onChange={handleChange}
      >
        {sortByOptions.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
};
