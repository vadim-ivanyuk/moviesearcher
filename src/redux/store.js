import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./rootReducer";
import { ON_CHANGE_FILTERS } from "./filters/filters.types";

let currentPage;

const onChangePageMiddleware = ({ dispatch, getState }) => (next) => (
  action
) => {
  currentPage = getState().filters.page;
  if (
    action.type === ON_CHANGE_FILTERS &&
    action.payload.name !== "page" &&
    currentPage !== 1
  ) {
    dispatch({
      type: ON_CHANGE_FILTERS,
      payload: {
        name: "page",
        value: 1,
      },
    });
  }

  return next(action);
};

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, onChangePageMiddleware))
);

export default store;
