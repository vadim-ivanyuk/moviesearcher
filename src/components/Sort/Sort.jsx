import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { onChangeFilters } from "../../redux";

const sortByOptions = [
  {
    label: "Популярные (убывание)",
    value: "popularity.desc",
  },
  {
    label: "Популярные (возростание)",
    value: "popularity.asc",
  },
  {
    label: "Рейтинг (убывание)",
    value: "vote_average.desc",
  },
  {
    label: "Рейтинг (возростание)",
    value: "vote_average.asc",
  },
];

export const Sort = () => {
  const filters = useSelector((store) => store.filters);
  const dispatch = useDispatch();

  const handleChange = (e) => {
    const { name, value } = e.target;
    dispatch(onChangeFilters({ name, value }));
  };

  return (
    <div className="sort form-group">
      <label className="filters-label" htmlFor="sort_by">
        Сортировать
      </label>
      <select
        className="form-control cursor-pointer"
        name="sort_by"
        id="sort_by"
        onChange={handleChange}
        value={filters.sort_by}
      >
        {sortByOptions.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
};
