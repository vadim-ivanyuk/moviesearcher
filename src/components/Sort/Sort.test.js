import React from "react";
import { configure, render } from "enzyme";
import EnzymeToJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../redux/store";
import { Sort } from "./Sort";

configure({ adapter: new Adapter() });

describe("Sort", () => {
  it("snap Sort", () => {
    const sortRender = render(
      <Provider store={store}>
        <Sort />
      </Provider>
    );
    expect(EnzymeToJSON(sortRender)).toMatchSnapshot();
  });
});
