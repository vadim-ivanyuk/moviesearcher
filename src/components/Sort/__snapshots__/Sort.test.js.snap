// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort snap Sort 1`] = `
<div
  class="sort form-group"
>
  <label
    class="filters-label"
    for="sort_by"
  >
    Сортировать
  </label>
  <select
    class="form-control cursor-pointer"
    id="sort_by"
    name="sort_by"
  >
    <option
      value="popularity.desc"
    >
      Популярные (убывание)
    </option>
    <option
      value="popularity.asc"
    >
      Популярные (возростание)
    </option>
    <option
      value="vote_average.desc"
    >
      Рейтинг (убывание)
    </option>
    <option
      value="vote_average.asc"
    >
      Рейтинг (возростание)
    </option>
  </select>
</div>
`;
